pulp_smash/tests/pulp3/file/api_v3/test_auto_distribution.py

```diff
@@ -0,0 +1,137 @@
+
+from pulp_smash import api, config, utils
+from pulp_smash.constants import FILE_FEED_URL, FILE_URL
+from pulp_smash.tests.pulp3.constants import (
+    DISTRIBUTION_PATH,
+    FILE_CONTENT_PATH,
+    FILE_PUBLISHER_PATH,
+    FILE_REMOTE_PATH,
+    REPO_PATH,
+)
+from pulp_smash.tests.pulp3.file.api_v3.utils import gen_publisher, gen_remote
+from pulp_smash.tests.pulp3.file.utils import set_up_module as setUpModule  # pylint:disable=unused-import
+from pulp_smash.tests.pulp3.pulpcore.utils import gen_distribution, gen_repo
+from pulp_smash.tests.pulp3.utils import (
+    get_added_content,
+    get_auth,
+    get_versions,
+    publish,
+    sync,
+)
+
+
+class AutoDistributionTestCase(unittest.TestCase):
+    """Test auto distribution."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables.
+
+        Add content to Pulp.
+        """
+        cls.cfg = config.get_config()
+        cls.client = api.Client(cls.cfg, api.json_handler)
+        cls.client.request_kwargs['auth'] = get_auth()
+        remote = {}
+        repo = {}
+        try:
+            body = gen_remote(urljoin(FILE_FEED_URL, 'PULP_MANIFEST'))
+            remote.update(cls.client.post(FILE_REMOTE_PATH, body))
+            repo.update(cls.client.post(REPO_PATH, gen_repo()))
+            sync(cls.cfg, remote, repo)
+            cls.content = cls.client.get(FILE_CONTENT_PATH)['results']
+        finally:
+            if remote:
+                cls.client.delete(remote['_href'])
+            if repo:
+                cls.client.delete(repo['_href'])
+
+    def test_repo_auto_distribution(self):
+        """Test auto distribution of a repository.
+
+        This test targets the following issue:
+
+        * `Pulp Smash #947 <https://github.com/PulpQE/pulp-smash/issues/947>`_
+
+        Do the following:
+
+        1. Create a repository that has at least one repository version.
+        2. Create a publisher.
+        3. Create a distribution and set the repository and publishera to the
+           previous created ones.
+        4. Create a publication using the latest repository version.
+        5. Assert that the previous distribution has a  ``publication`` set as
+           the one created in step 4.
+        6. Create a new repository version by adding content to the repository.
+        7. Create another publication using the just created repository
+           version.
+        8. Assert that distribution now has the ``publication`` set to the
+           publication created in the step 7.
+        9. Verify that content added in the step 7 is now available to download
+           from distribution, and verify that the content unit has the same
+           checksum when fetched directly from Pulp-Fixtures.
+        """
+        # Create a repository.
+        repo = self.client.post(REPO_PATH, gen_repo())
+        self.addCleanup(self.client.delete, repo['_href'])
+        for content in self.content[:-1]:
```

Comment:
Is there any special reason to iterate through all existing content, and to create a new repository version for each individual piece of content? This list can be quite long, depending on what prior test cases have executed. `self.content` could be as little as two elements long, and this test will still work.